cumulusci/util/xml/metadata_tree.py

```diff
@@ -0,0 +1,222 @@
+    """
+
+    doc = etree.parse(source)
+    return MetadataElement(doc.getroot())
+
+
+def fromstring(source):
+    """Parse a Metadata Tree from a string"""
+    return MetadataElement(etree.fromstring(source))
+
+
+class MetadataElement:
+    '''A class for representing Metadata in a Pythonic tree.
+
+    After you parse into a MetadataElement tree, you can refer
+    to child elements like this:
+
+    >>> from cumulusci.util.xml import metadata_tree
+    >>> Package.types.name.text
+    'StandardValueSet'
+    >>> Package = metadata_tree.fromstring(b"""<?xml version="1.0" encoding="UTF-8"?>
+    ... <Package xmlns="http://soap.sforce.com/2006/04/metadata">
+    ...     <types>
+    ...         <members>CaseReason</members>
+    ...         <members>CaseStatus</members>
+    ...         <members>CaseType</members>
+    ...         <name>StandardValueSet</name>
+    ...     </types>
+    ...     <version>46.0</version>
+    ... </Package>""")
+    >>> Package.types.name.text
+    'StandardValueSet'
+
+    Or you can refer to them like this:
+
+    >>> Package["types"]["name"].text
+    'StandardValueSet'
+
+    That might be convenient if you had the key names in a variable or if you had to refer to an element name
+    which would clash with a Python keyword (like `import`) or an instance method (like `text`).
+
+    You can refer to members of lists by index:
+
+    >>> Package["types"]["members"][1]
+    <members>CaseStatus</members> element
+
+    There are also methods for finding, appending, inserting and removing nodes, which have their own documentation.
+    '''
+
+    __slots__ = ["_element", "_parent", "_ns", "tag"]
+
+    def __init__(
+        self, element: etree._Element, parent: etree._Element = None,
+    ):
+        assert isinstance(element, etree._Element)
+        assert len(element.nsmap) == 1, "Only one namespace allowed"
+        self._element = element
+        self._parent = parent
+        self._ns = next(iter(element.nsmap.values()))
+        self.tag = element.tag.split("}")[1]
+
+    @property
+    def text(self):
+        if len(self._element):
+            text_node = self._element.find(self._add_namespace("text"))
+            if len(text_node) >= 0:
+                return self._wrap_element(text_node)
+        return self._element.text
+
+    @text.setter
+    def text(self, text):
+        self._element.text = text
+
+    def _wrap_element(self, child: etree._Element):
+        return MetadataElement(child, self._element)
+
+    def _add_namespace(self, tag):
+        return "{%s}%s" % (self._ns, tag)
+
+    def _get_child(self, childname):
+        child_element = self._element.find(self._add_namespace(childname))
```

Comment:
This is an internal function. 

The indexing occurs here:

https://github.com/SFDO-Tooling/CumulusCI/pull/1560/files#diff-0f00573b2fbf4af992b4caf5820f88a2R116

And as I mentioned elsewhere, it does this because

x.y

Needs to be ready to behave as the base stem of either:

x.y.z

Or

x.y[3]

In other words it behaves as either a list or an element depending on what happens 
"next". The base-case is element so I need an element that can behave as a list.